sklearn/utils/estimator_checks.py

```diff
@@ -1475,22 +1475,25 @@ def check_estimators_overwrite_params(name, estimator_orig):
 
 
 @ignore_warnings(category=(DeprecationWarning, FutureWarning))
-def check_no_fit_attributes_set_in_init(name, Estimator):
-    """Check that Estimator.__init__ doesn't set trailing-_ attributes."""
-    # this check works on classes, not instances
+def check_no_attributes_set_in_init(name, Estimator):
+    """Check that Estimator.__init__ doesn't set any attributes apart from
+       the init parameters.
+    """
     estimator = Estimator()
-    for attr in dir(estimator):
-        if attr.endswith("_") and not attr.startswith("__"):
-            # This check is for properties, they can be listed in dir
-            # while at the same time have hasattr return False as long
-            # as the property getter raises an AttributeError
-            assert_false(
-                hasattr(estimator, attr),
-                "By convention, attributes ending with '_' are "
-                'estimated from data in scikit-learn. Consequently they '
-                'should not be initialized in the constructor of an '
-                'estimator but in the fit method. Attribute {!r} '
-                'was found in estimator {}'.format(attr, name))
+    if(name not in ["GaussianProcess", "RandomizedLasso",
```

Comment:
I don't know if I have the wrong config but these classes (GaussianProcess, RandomizedLasso... ) don't have `deprecated_original` or `deprecated` attribute ?